extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -119,17 +119,18 @@ public OidcTenantConfig apply(OidcTenantConfig tenantConfig) {
                             final String tenantId = context.get(OidcUtils.TENANT_ID_ATTRIBUTE);
 
                             if (tenantId != null && !isTenantSetByAnnotation(context, tenantId)) {
-                                TenantConfigContext tenantContext = tenantConfigBean.getDynamicTenantsConfig().get(tenantId);
+                                // WARN: The order (check dynamic before static) is important!
+                                var tenantContext = tenantConfigBean.getDynamicTenant(tenantId);
                                 if (tenantContext != null) {
                                     // Dynamic map may contain the static contexts initialized on demand,
-                                    if (tenantConfigBean.getStaticTenantsConfig().containsKey(tenantId)) {
+                                    if (tenantConfigBean.getStaticTenant(tenantId) != null) {
```

Comment:
Please remove the comment above since it is no longer relevant

Reply:
#43110 removes it (the comment's still valid for this PR)